docs/source/ddp_comm_hooks.rst

```diff
@@ -0,0 +1,64 @@
+DDP Communication Hooks
+=======================
+
+DDP communication hook is a generic interface to control how to communicate
+gradients across workers.
+A few built-in communication hooks are provided,
+and users can easily apply any of these hooks to optimize communication.
+Besides, the hook interface can also support user-defined communication
+strategies for more advanced use cases.
+
+.. warning ::
+    DDP communication hooks can only support single process single device mode
```

Comment:
As a side note, may be good to double check that we throw the appropriate errors if wrong backend or mode is used.

Reply:
1) single-process single-device mode is already checked in `reducer.cpp`.
2) will add another check for the backend in a separate PR.